Oracle.md

```diff
@@ -0,0 +1,135 @@
+# Oracle specifications
+
+## Introduction
+
+For the purpose of these specifications, an event is a digital representation of a real work fact.
+An [oracle](./Introduction#Oracle) is an entity that commits to publishing a signature over a (number of) event outcome(s) ahead of time by releasing an [R-value](./Introduction#R-value) as well as necessary information for two parties to build a set of [CETs](./Introduction#Contract-Execution-Transaction-(CET)).
+This information is committed in a so-called _event template_ that will be further detailed in this document.
+Templates are classified and discovered through a namespace presently represented as Unique Resource Identifier (URI) paths.
+
+## Event templates
+
+An event template provides information to clients about an event for which an oracle plans on releasing a signature over its outcome.
+The provided information should be sufficient for a client to create a set of CETs that will cover all possible outcomes of the event.
+
+Regardless of the outcome type, an event template should include:
+
+- The R value that will be used to sign the event outcome (possibly several).
+- The public key of the oracle.
+- The earliest date at which the event outcome will be released in Unix epoch format.
+
+Here we assume that an event outcome can be represented either as a set of strings or numbers.
+Three kinds of event outcomes are defined (note that a single even can be composed of several types).
+
+### Simple enumeration
+
+For events that have a narrow range of possible outcomes, the possible outcomes can simply be enumerated.
+For the sake of simplicity, outcomes should be represented as strings.
+
+#### Example: Weather tomorrow
+
+Tomorrow's weather can be represented as the set of strings `[sunny, cloudy, rainy]`.
```

Comment:
Where does the information go about which location this is at and whose tomorrow it is? You don't just sign `sunny` right? You must sign something like "It was sunny in Paris on 2020-08-15" or am I missing something.

Reply:
For me the "in paris on 2020-08-15" should be encoded in the URI. So you would retrieve the information from "/weather/france/paris/2020-08-15" for example and then the signature would indeed just be over "sunny". (And I realize that the URI should probably be part of what the oracle commit to prior to the event).

Reply:
Specified that the URI should be included in the attestation signature.

Reply:
Wait so you decided to put it in the attestation after all? i.e. not just sign "sunny". I think that would be better otherwise in order for a system to know the outcome of an event they would not be able to tell just from the signature and the message.

Reply:
What I referred as attestation signature is not the signature over the event outcome but the commitment of the oracle to provide the event outcome in the future (sorry if I got the terms wrong)

Reply:
The terms I am using are:
1. announcement: When the oracle announces its intention to attest to an event outcome (provides R values)
2. attestation: The decision on the outcome through revealing the corresponding signature scalar.

It is sub-optimal to sign just "sunny" since, given the signature from the attestation you cannot tell what fact the oracle was attesting to just that the outcome is "sunny". You would have to combine this with the announcement signature to infer what the actual fact was. In my vision oracles are also useful as a source of historical facts rather than just DLCs.

Reply:
Thanks I'll try to follow this as well then, I like this terminology.

I don't have strong opinion about what is being signed so I'm ok with adding some extra data to it.